Tests/Common/SimulatedBasicsStation.cs

```diff
@@ -0,0 +1,176 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaWan.Tests.Common
+{
+
+    using System;
+    using System.Text.Json;
+    using LoRaWan.Tests.Simulation.Models;
+    using Websocket.Client;
+    using System.IO;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using System.Runtime.InteropServices;
+    using LoRaWan.NetworkServer;
+    using LoRaTools.LoRaMessage;
+    using LoRaTools.Utils;
+    using System.Net.WebSockets;
+    using Xunit;
+    using System.Linq;
+
+    public sealed class SimulatedBasicsStation : IDisposable
+    {
+        private readonly string stationEUI;
+        private WebsocketClient DataWebsocketClient { get; set; }
+        private Uri LnsUri { get; set; }
+
+        private readonly HashSet<Func<string, bool>> subscribers = new HashSet<Func<string, bool>>();
+
+        internal void SubscribeOnce(Func<string, bool> value)
+        {
+            this.subscribers.Add(value);
+        }
+
+
+        public SimulatedBasicsStation(string stationEUI, Uri lnsUri)
+        {
+            this.stationEUI = stationEUI;
+            this.LnsUri = lnsUri;
+        }
+
+        public async Task StartAsync()
+        {
+            try
+            {
+                var routerReceivedMessage = new List<string>();
+                using var routerWebsocketClient = new WebsocketClient(new Uri(LnsUri, "router-info"))
```

Comment:
This code here does the initial handshake by the basicstation, I realized it is unnessary at the moment as we don't formaly check a basicstation went through it to accept its messages. still it seems to do the job.